File: app/src/main/java/fergaral/datetophoto/services/ProcessPhotosService.java
package fergaral.datetophoto.services;

import android.app.IntentService;
import android.content.Intent;
import android.os.ResultReceiver;
import android.support.annotation.Nullable;

import java.util.ArrayList;

import fergaral.datetophoto.R;
import fergaral.datetophoto.utils.NotificationUtils;
import fergaral.datetophoto.utils.ProcessPhotos;

public class ProcessPhotosService extends IntentService {

    private static boolean mRunning;

    public ProcessPhotosService() {
        super(ProcessPhotosService.class.getSimpleName());
    }

    @Override
    protected void onHandleIntent(@Nullable Intent intent) {
        try {
            startForeground(NotificationUtils.NOTIFICATION_ID,
                    new NotificationUtils(this).showProgressNotification("Procesando fotos..."));
            mRunning = true;
            ResultReceiver receiver = intent.getParcelableExtra("receiver");
            boolean onBackground = intent.getBooleanExtra("onBackground", true);
            ArrayList<String> cameraImages = intent.getStringArrayListExtra("cameraimages");
            new ProcessPhotos().execute(receiver, onBackground, cameraImages, this);
        } finally {
            stopForeground(true);
            String notificationText = getString(R.string.process_has_finished);
            new NotificationUtils(this).setUpNotification(2, false, false, notificationText, notificationText);
            mRunning = false;
        }
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        mRunning = false;
    }

    public static boolean isRunning() {
        return mRunning;
    }
}

File: app/src/main/java/fergaral/datetophoto/services/ProcessPhotosService.kt
package fergaral.datetophoto.services

import android.app.IntentService
import android.content.Intent
import android.os.ResultReceiver

import java.util.ArrayList

import fergaral.datetophoto.R
import fergaral.datetophoto.utils.NotificationUtils
import fergaral.datetophoto.utils.ProcessPhotos

class ProcessPhotosService : IntentService(ProcessPhotosService::class.java!!.getSimpleName()) {

    override fun onHandleIntent(intent: Intent?) {
        try {
            startForeground(NotificationUtils.NOTIFICATION_ID,
                    NotificationUtils(this).showProgressNotification("Procesando fotos..."))
            isRunning = true
            val receiver = intent!!.getParcelableExtra<ResultReceiver>("receiver")
            val onBackground = intent.getBooleanExtra("onBackground", true)
            val cameraImages = intent.getStringArrayListExtra("cameraimages")
            ProcessPhotos().execute(receiver, onBackground, cameraImages, this)
        } finally {
            stopForeground(true)
            val notificationText = getString(R.string.process_has_finished)
            NotificationUtils(this).setUpNotification(2, false, false, notificationText, notificationText)
            isRunning = false
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        isRunning = false
    }

    companion object {

        var isRunning: Boolean = false
            private set
    }
}
